plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,265 @@
+			tempSpace := info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor.(map[string]interface{})
+			tags["temp_path"] = tempSpace["path"].(string)
+			fields["temp_available"] = tempSpace["size"].(float64)
+		}
+
+		if info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor != nil {
+			swapSpace := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor.(map[string]interface{})
+			fields["swap_available"] = swapSpace["availableSwapSpace"].(float64)
+			fields["swap_total"] = swapSpace["totalSwapSpace"].(float64)
+			fields["memory_available"] = swapSpace["availablePhysicalMemory"].(float64)
+			fields["memory_total"] = swapSpace["totalPhysicalMemory"].(float64)
+		}
+
+		acc.AddFields(measurement, fields, tags)
+
+	}
+
+	return nil
+}
+
+func (j *Jenkins) GetJobsData(jobs []*gojenkins.Job, acc telegraf.Accumulator) error {
+
+	measurement := "jenkins_job"
+
+	for _, job := range jobs {
+
+		jobName := job.GetName()
+
+		for _, filtername := range j.JobFilterName {
+			if filtername == jobName {
+				continue
+			}
+		}
+
+		// ignore if job has no builds
+		if job.Raw.LastBuild.Number < 1 {
+			continue
+		}
+
+		jobLastBuild, err := job.GetLastBuild()
+		if err != nil {
+			log.Printf("E! Error retrieving last build from job %s: %s", jobName, err.Error())
+			return err
+		}
+
+		// ignore if last build is too old
+		if (j.LastbuildFilterInterval != internal.Duration{Duration: 0}) {
+			buildSecAgo := time.Now().Sub(jobLastBuild.GetTimestamp()).Seconds()
+
+			if buildSecAgo > j.LastbuildFilterInterval.Duration.Seconds() {
+				log.Printf("D! Last job too old, last %s build was %v seconds ago", jobName, buildSecAgo)
+				continue
+			}
+		}
+
+		buildIds, err := job.GetAllBuildIds()
+		sort.Sort(byBuildNumber(buildIds))
+
+		if err != nil {
+			log.Printf("E! Error retrieving all builds from job \"%s\": %s", jobName, err.Error())
+			return err
+		}
+
+		for _, buildId := range buildIds {
+			build, err := job.GetBuild(buildId.Number)
+			log.Printf("D! Reading data from job %s build %v", jobName, buildId.Number)
+			if err != nil {
+				log.Printf("E! Error retrieving build from job \"%s\": %s", jobName, err.Error())
+				return err
+			}
+
+			// ignore if build is ongoing
+			if build.IsRunning() {
+				log.Printf("D! Ignore running build on %s, build %v", jobName, buildId.Number)
+				continue
+			}
+
+			// stop if build is too old
+			if (j.LastbuildFilterInterval != internal.Duration{Duration: 0}) {
+				buildSecAgo := time.Now().Sub(build.GetTimestamp()).Seconds()
+
+				if time.Now().Sub(build.GetTimestamp()).Seconds() > j.LastbuildFilterInterval.Duration.Seconds() {
+					log.Printf("D! Job %s build %v too old (%v seconds ago), skipping to next job", jobName, buildId.Number, buildSecAgo)
+					break
+				}
+			}
+
+			tags := map[string]string{"job_name": jobName, "result": build.GetResult()}
+			fields := make(map[string]interface{})
+			fields["duration"] = build.GetDuration()
```

Comment:
What are the units?  Let's name it with the units, ex: `duration_ms`

Reply:
Yes, it is milliseconds. That is a breaking change for me, I will need to see how I can change my data in InfluxDB :)

Reply:
It might be helpful to look at this [migration guide for the mysql plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mysql#metric-version), you could change to the TICKScript to (untested):
```
batch
    |query('''
        SELECT duration as duration_ms FROM "telegraf"."autogen"."jenkins"
    ''')
        .period(5m)
        .every(5m)
	|influxDBOut()
		.database('telegraf')
		.retentionPolicy('autogen')
```